src/overlays/actors/ovl_Boss_Goma/z_boss_goma.c

```diff
@@ -21,115 +111,1836 @@ const ActorInit Boss_Goma_InitVars = {
+    0x00, 0x00,
+    0x1F, // part of right arm (shell on shoulder)
+    0x23, // part of right arm (shoulder)
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x2B, // end of left antenna
+    0x30, // middle of left antenna
+    0x35, // start of left antenna
+    0x00, 0x00, 0x00, 0x00,
+    0x2A, // end of right antenna
+    0x2D, // middle of right antenna
+    0x35, // start of right antenna
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x0B, // back of left claw/hand
+    0x0F, // front of left claw/hand
+    0x15, // part of left arm (inner)
+    0x00, 0x00,
+    0x19, // part of left arm (shell)
+    0x00, 0x00,
+    0x1E, // part of left arm (shell on shoulder)
+    0x23, // part of left arm (shoulder)
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+};
+
+/**
+ * Sets the `i`th pixel of a 16x16 RGBA16 image to 0 (transparent black)
+ * according to the `clearPixelTable`
+ */
+void BossGoma_ClearPixels16x16Rgba16(s16* rgba16image, u8* clearPixelTable, s16 i) {
+    if (clearPixelTable[i]) {
+        rgba16image[i] = 0;
+    }
+}
+
+/**
+ * Sets the `i`th 2x2 pixels block of a 32x32 RGBA16 image to 0 (transparent black)
+ * according to the `clearPixelTable`
+ */
+void BossGoma_ClearPixels32x32Rgba16(s16* rgba16image, u8* clearPixelTable, s16 i) {
+    s16* targetPixel;
+
+    if (clearPixelTable[i]) {
+        // address of the top left pixel in a 2x2 pixels block located at
+        // (i & 0xF, i >> 4) in a 16x16 grid of 2x2 pixels
+        targetPixel = (s32)rgba16image + (s16)((i & 0xF) * 2 + (i & 0xF0) * 4) * 2;
+        // set the 2x2 block of pixels to 0
+        targetPixel[0] = 0;
+        targetPixel[1] = 0;
+        targetPixel[32 + 0] = 0;
+        targetPixel[32 + 1] = 0;
+    }
+}
+
+/**
+ * Clear pixels from Gohma's textures
+ */
+void BossGoma_ClearPixels(u8* clearPixelTable, s16 i) {
+    // rgba16 16x16
+    BossGoma_ClearPixels16x16Rgba16((s16*)SEGMENTED_TO_VIRTUAL(D_060183A8), clearPixelTable, i);
+    BossGoma_ClearPixels16x16Rgba16((s16*)SEGMENTED_TO_VIRTUAL(D_060185A8), clearPixelTable, i);
+    BossGoma_ClearPixels16x16Rgba16((s16*)SEGMENTED_TO_VIRTUAL(D_060187A8), clearPixelTable, i);
+    BossGoma_ClearPixels16x16Rgba16((s16*)SEGMENTED_TO_VIRTUAL(D_060191A8), clearPixelTable, i);
+    // rgba16 32x32
+    BossGoma_ClearPixels32x32Rgba16((s16*)SEGMENTED_TO_VIRTUAL(D_060189A8), clearPixelTable, i);
+    BossGoma_ClearPixels32x32Rgba16((s16*)SEGMENTED_TO_VIRTUAL(D_060193A8), clearPixelTable, i);
+}
+
+void BossGoma_Init(Actor* thisx, GlobalContext* globalCtx) {
+    static InitChainEntry D_8091B2A8[] = { ICHAIN_U8(unk_1F, 2, ICHAIN_CONTINUE),
+                                           ICHAIN_S8(naviEnemyId, 1, ICHAIN_CONTINUE),
+                                           ICHAIN_F32_DIV1000(gravity, -2000, ICHAIN_STOP) };
```

Comment:
we're gonna leave ichains outside of the function for now, so you can leave it right above
also add a trailing comma to the last entry so clang doesnt do the dumb indenting